dev/TestMailer.php

```diff
@@ -4,14 +4,14 @@
  * @subpackage email
  */
 class TestMailer extends Mailer {
-	protected $emailsSent = array();
+	protected static $emailsSent = array();
```

Comment:
I investigated a bit further. It seems like using `Email::mailer()` will get you the correct instance, while `SapphireTest::$mailer` will return another (unpopulated) instance. That's really strange, since SapphireTest sets `Injector::inst()->registerService($this->mailer, 'Mailer');` in the startup, which should make them the same instance.

I guess it's time to start the debugger to see what's going on.